.github/workflows/ci.yml

```diff
@@ -23,39 +24,70 @@ jobs:
     env:
       GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     outputs:
-      device-db: ${{ steps.load-db.outputs.device-db }}
-      device-list: ${{ steps.load-db.outputs.device-list }}
+      config-path: ${{ steps.load-config.outputs.config-path }}
+      device-list: ${{ steps.load-config.outputs.device-list }}
       magisk-key: ${{ steps.get-magisk-cache.outputs.magisk-key }}
-      magisk-url: ${{ steps.load-db.outputs.magisk-url }}
-      hit-img: ${{ steps.get-img-cache.outputs.img-hit }}
-      hit-tox: ${{ steps.get-tox-cache.outputs.tox-hit }}
+      img-key-prefix: ${{ steps.get-img-cache.outputs.img-key-prefix }}
+      img-hit: ${{ steps.get-img-cache.outputs.img-hit }}
+      tox-key-prefix: ${{ steps.get-tox-cache.outputs.tox-key-prefix }}
+      tox-hit: ${{ steps.get-tox-cache.outputs.tox-hit }}
     steps:
       - uses: actions/checkout@v3
+        with:
```

Comment:
Do we really need submodules in this job?

Reply:
As `tests.py` is currently written, yes, due to the global import of OTA-related modules. Python used to do lazy imports, but doesn't anymore (and I think the PEP-690 proposal to add it back was rejected).

Maybe I can move the OTA-related import to `strip_image()`.

Reply:
I am not all that familiar with packaging in Python buy we only import tests/config.py right? That file doesn't have any dependency on the submodule. Or does this also import tests/tests py in some way?

Reply:
Yep, that's correct. The Python script inside the GA workflow doesn't require it, but `preload-magisk-cache` does because it uses `./tests/tests.py download <...>` now.